### ISSUE TITLE
PHP 8.2 issue in DisableConstructorPatch

### ISSUE THREAD

**Issue by lolli42** [phpspec/prophecy#555]
```
        $constructor->setCode(<<<PHP
if (0 < func_num_args()) {
    call_user_func_array(array('parent', '__construct'), func_get_args());
}
PHP
        );
```

This raises `Use of "parent" in callables is deprecated` E_DEPRECATED due to https://wiki.php.net/rfc/deprecate_partially_supported_callables .

**Comment by ciaranmcnulty** [comment]
Thanks for reporting; I guess we'll have to use the real class name?

**Comment by gilbertsoft** [comment]
useParentCode() does exactly what should be done here.

**Comment by TalalM** [comment]
Hello,

What's the status of this fix?
I am moving my project to PHP 8.2 and my tests are failing now because of this error.

Will this be fixed, or is there a way for me to make my tests pass?

Thanks,

**Comment by stof** [comment]
I just merged #614 which provided a fix.

**Comment by TalalM** [comment]
OK thank you.

Do you know when is the next release, in which the fix will be available?
